### PR TITLE
Galaxy app integrations

### LINKED ISSUE #45: Figure out a way for other galaxy apps to change messaging based on context
- Let other galaxy apps change strings for login, signup, verification, etc to something custom.
- Create a dictionary (map, type, whatever) that maps specific messaging to the actual string (e.g login_subtitle_text => "Login to continue to your TD 2020 Application")
- Maybe make a service that takes the redirect route e.g "/apply/*" and maps it to a dictionary that has the custom strings
- Then each view would take this dictionary and use those strings where they apply.

### PR DESCRIPTION
**Pull request by phulsechinmay**
Closes #45 

**Features:**
* Added a `GalaxyIntegrationConfig` interface that specifies the UI / messaging overrides that a Galaxy app can define
* Added a function that returns a config with default messaging for overrides that don't want to be explicitly defined
* Created a decorator that returns the appropriate integration config according to the redirectPath specified in the request query params
  * Uses longest matching prefix to find the config to use

### CODE REVIEW

**Comment by phulsechinmay** [comment]
@geooot Just needed to get some ideation on this. Currently all the Galaxy app integrations are defined under a constants file. If we're going to open this up so messaging is stored in some database and can be edited by hand / through some defined endpoints, then this PR will quite a few changes.

However, if we're good with them being defined in a constants file right now, I've also added a decorator that returns the config according to the redirect path.

Something to consider would be that Gatekeeper would need to be redeployed everytime we change this constants file.

**Comment by geooot** [comment]
@phulsechinmay I think the strings should just be in a constants file, no need to store it in a database

File: src/galaxy-integrations/constants/default-integrations.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { newGalaxyIntegrationConfig } from "../interfaces/galaxy-intergration";
```
Comment (geooot) [inline src/galaxy-integrations/constants/default-integrations.ts:1]: Make spelling of `galaxy-intergtation` correct

File: src/galaxy-integrations/interfaces/galaxy-intergration.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  appId?: string;
+
+  /* Login Page Variables */
+  loginPageSubtext?: string;
+
+  /* Signup Page Variables */
+  signupPageSubtext?: string;
+}
+
+export const newGalaxyIntegrationConfig = (
```
Comment (geooot) [inline src/galaxy-integrations/interfaces/galaxy-intergration.ts:18]: Add a JSDoc for this

File: src/galaxy-integrations/decorators/redirect-galaxy-integration.decorator.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { createParamDecorator, ExecutionContext } from "@nestjs/common";
+import {
+  DefaultGalaxyIntegrations,
+  GatekeeperGalaxyIntegration
+} from "../constants/default-integrations";
+
+const getLongestMatchingPrefix = (key, prefixes) => {
```
Comment (geooot) [inline src/galaxy-integrations/decorators/redirect-galaxy-integration.decorator.ts:7]: Add a JSDoc

**Comment by phulsechinmay** [comment]
@geooot I added the test for the `getLongestPathPrefix` function and fixed it to be more logically correct. About the test for the decorator, seems like it's a pain to test decorators in NestJS ([source](https://medium.com/@enrico.ferro/create-a-test-case-for-custom-parameter-decorator-in-nestjs-7-d97de5c4e6a0)). 

Looking at that, do you think a test for it is worth it?

**Pull request merged**